### ISSUE TITLE
[BUG] `T.__tan` on `float32` silently returns inaccurate values

### ISSUE THREAD

**Issue by Dino1844** [tile-ai/tilelang#2879]
### Required prerequisites

- [x] I have read the documentation <https://tilelang.com>.
- [x] I have searched the [Issue Tracker](https://github.com/tile-ai/tilelang/issues) that this hasn't already been reported. (comment there if it has.)

### What version of TileLang are you using?

0.1.13

### System information

Python 3.12.3, torch 2.12.1+cu130, CUDA 13.2 (nvcc 13.2.78); On NVIDIA A100 (sm_80).

### Problem description

`T.__tan(x)` on `float32` silently returns values off by orders of magnitude, instead of a bounded fast-math approximation like the rest of the `__` family (`__sinf`/`__cosf`/`__expf` all match torch to ~1e-6). `__tanf`'s error is unbounded near the poles. 200k `randn` points vs `torch.tan`:

| call | max abs err | mean | err > 1e-3 |
|---|---|---|---|
| `__tanf` (current) | **1791.3** | 1.2e-2 | 861 / 200000 |
| `tanf` (intended) | 0 | 0 | 0 |

### Reproducible example code

The Python snippets:

```python
import torch, tilelang
import tilelang.language as T

@tilelang.jit(out_idx=[1])
def tan_kernel(N: int):
    @T.prim_func
    def main(A: T.Tensor((N,), "float32"), B: T.Tensor((N,), "float32")):
        with T.Kernel(1, threads=128):
            for i in T.Parallel(N):
                B[i] = T.__tan(A[i])   # emits __tanf(...)
    return main

a = torch.randn(1024, device="cuda", dtype=torch.float32)
print((tan_kernel(1024)(a) - torch.tan(a)).abs().max().item())
```

### Traceback

```pytb
None — compiles and runs, results silently wrong and deterministic.
```

### Expected behavior

`T.__tan` on fp32 should satisfy the same accuracy bar the repo's fastmath tests apply to `__exp`/`__log` (`torch.testing.assert_close(..., rtol=1e-3, atol=1e-3)`), or fall back to accurate `tanf` per the project's own documented intent (see below) — not silently return values orders of magnitude off near the poles.

### Additional context

Root cause: `tl.__tan` uses `CUDAFastMath` → `__tanf` (`src/cuda/codegen/codegen_cuda.cc:4618-4622`). The intended fix `CUDAFastMathTan` (fp32 → `tanf`, with the comment "`__tanf` ... too deviant from numpy tan ... use just `tanf`") is **dead code** at `codegen_cuda.cc:175-193` and `intrin_rule_cuda.cc:72-90` — ROCm wires its equivalent (`HIPFastMathTan`, `intrin_rule_hip.cc:223`), CUDA forgot. Same mapping also makes fp16/bf16 `tan` emit non-existent `htan` (only `htanh` exists) → compile failure.

Tests stay green because `run_fastmath_mathop_test` compares `__tan` against `expected = b_fastmath` (self-comparison); only `__exp`/`__log` are validated vs torch.
